### ISSUE TITLE
Option to treat function calls in value bindings as computed properties

### ISSUE THREAD

**Issue by seaskyways** [vuejs/vue#9661]
### What problem does this feature solve?
This feature gives a performance boost for applications such as translation bindings as it reduces the number of redundant function calls. 

### What does the proposed API look like?
Adding an option with "." (like ".sync") to property binding called ".computed" as such.

```html
<template>
  <Modal :title.computed="$t('MY_TITLE_KEY')"/> 
</template>
```

Which would then be kind of translated to:

```html
<template>
  <Modal :title="randomComputed1"/> 
</template>

<script>
export default {
  computed: {
    randomComputed1() { return this.$t('MY_TITLE_KEY') }
  }
}
<script>
```

**Comment by posva** [comment]
I don't think this is possible in current version of Vue as it would require to add something to the component options. However, I wonder if it could be made possible with hooks in Vue 3

**Comment by DaKoala** [comment]
Agree with @posva. I think in this case the Vue instance is modified by the template.

**Comment by seaskyways** [comment]
The Vue instance is already being modified by the template. The template is being compiled to a render function which is being added to the Vue component object.
